.changeset/light-turtles-yawn.md

```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-catalog-backend': patch
+---
+
+Removed deprecated types that are exported in plugins/catalog-backend/src/ingestion/index.ts. Also fixed deprecated use of substr into substring in the files plugins/catalog-backend/src/modules/core/PlaceholderProcessor.ts and plugins/catalog-backend/src/service/request/parseEntityFilterParams.ts.
```

Comment:
Remember that this file's contents is what ends up in the public changelog. It should be written with an external consumer of the library in mind, not the reviewer. As such, deep details about filenames and non-public concerns should be left out. This changeset could instead be:

`Internal refactoring to avoid use of deprecated functions.`